fastparse/shared/src/main/scala/fastparse/core/Parsing.scala

```diff
@@ -374,6 +374,26 @@ trait Parser[+T, ElemType, Repr] extends ParserResults[T, ElemType] with Precede
   }
 
   /**
+   * Extractor for pattern matching
+   *
+   *  For example:
+   *
+   *  {{{
+   *  val p1 = CharIn('a'to'z').! ~ CharIn('0'to'9').rep(1).!.map(_.toInt)
+   *  List("a42x") match {
+   *    case p1(x: String, y: Int) :: _ => // x is "a", y is 42
+   *  }
+   * }}}
+   */
+  def unapply[Repr <% IndexedSeq[ElemType]](input: Repr): Option[T] = {
```

Comment:
You can add any `Repr` type as you want by simply implementing new instance of `ResultConverter` and the case when we don't know the actual type of input data or these types are varying from run to run is not so big loss.